### ISSUE TITLE
Long Gif conversions

### ISSUE THREAD

**Issue by karaggeorge** [wulkano/kap#598]
[<img alt="Issuehunt badges" src="https://img.shields.io/badge/IssueHunt-%2480%20Funded-%2300A156.svg" />](https://issuehunt.io/r/wulkano/kap/issues/598)

**macOS version:** Any
**Kap version:** 3.x

#### Steps to reproduce
- Create a long recording
- Select only a small part (using trimmer handles)
- Export as Gif

#### Current behaviour
- The first process (palette) takes a long time
- The actual conversion is really fast

You can tell between the two because the palette has the indeterminate progress bar, while the conversion one has the normal one

#### Expected behaviour
- They should both be fast since the exported gif is tiny

#### Workaround
- None

The two ffmpeg calls are here: https://github.com/wulkano/kap/blob/master/main/convert.js#L133-L156

For the palette one we pass the entire recording. We might be able to pass the start/end times for that as well to make it faster

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $80.00)

- [<img src='https://avatars3.githubusercontent.com/u/44827199?v=4' alt='issuehunt' width=24 height=24> issuehunt](https://issuehunt.io/u/issuehunt) ($80.00)
### Submitted pull Requests
- [#694 Add start and end time for gif palette generation](https://issuehunt.io/r/wulkano/kap/pull/694)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/wulkano/kap/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

**Comment by sindresorhus** [comment]
Looks like we can just define starting point and duration like normal even for the palette generation: https://medium.com/@colten_jackson/doing-the-gif-thing-on-debian-82b9760a8483

**Comment by IssuehuntBot** [comment]
@issuehunt has funded $80.00 to this issue.

---
- Submit pull request via [IssueHunt](https://issuehunt.io/repos/65411043/issues/598) to receive this reward.
- Want to contribute? Chip in to this issue via [IssueHunt](https://issuehunt.io/repos/65411043/issues/598).
- Checkout the [IssueHunt Issue Explorer](https://issuehunt.io/issues) to see more funded issues.
- Need help from developers? [Add your repository](https://issuehunt.io/r/new) on IssueHunt to raise funds.

**Comment by topalavlad** [comment]
I've been using Kap for a while and really liked it so I thought I could try creating a PR.

I did some short tests and for a 1 minute recording where I trimmed the exported gif to only 2s, the paletteGeneration took ~18s for the old code and ~4s for the new one that uses start/end time to trim the input. 

Let me know if you think some other tests are required. I also think adding start/end time provides gets a better overall result since the palette is only generated for the selected output.
